.github/workflows/unit-tests.yaml

```
@@ -39,27 +39,16 @@ jobs:
         shell: "bash"
         run: "npm install -g @go-task/cli"
 
-      - if: "matrix.os == 'macos-latest'"
-        name: "Install coreutils (for md5sum)"
-        run: "brew install coreutils"
-
-      - if: "matrix.os == 'ubuntu-20.04'"
-        name: "Install gcc-10 (for c++20)"
-        run: |-
-          sudo apt-get update
-          sudo DEBIAN_FRONTEND=noninteractive apt-get install --no-install-recommends -y \
-              g++-10 \
-              gcc-10
-          sudo ln --symbolic --force /usr/bin/gcc-10 /usr/bin/cc
-          sudo ln --symbolic --force /usr/bin/g++-10 /usr/bin/c++
-          sudo ln --symbolic --force /usr/bin/cpp-10 /usr/bin/cpp
-
-      - name: "Log tool versions"
+      - if: "'macos-14' == matrix.os"
+        name: "Install MacOS-14 deps: coreutils (for md5sum) and AppleClang16 (for c++20)"
         run: |-
-          command -v md5sum
-          command -v python
-          command -v tar
-          command -v task
+          brew install coreutils
+          brew install llvm@16
 
       - name: "Run unit tests"
-        run: "task test-all"
+        env:
+          CC: "${{ 'macos-14' == matrix.os && '/opt/homebrew/opt/llvm@16/bin/clang' || '' }}"
+          CXX: "${{ 'macos-14' == matrix.os && '/opt/homebrew/opt/llvm@16/bin/clang++' || '' }}"
```

If I'm not mistaken, the current code will set `CC` and `CXX` when we're not running on macos-14, right? (We could add a test log to check.)

Although that works in this case, sometimes other tools will define `CC` and `CXX`, so I would prefer not to override them unless necessary. So maybe this is a better (albeit visually uglier) approach:

```suggestion
        env: >-
          ${{
          'macos-14' == matrix.os
          && fromJson('{
          "CC": "/opt/homebrew/opt/llvm@16/bin/clang",
          "CXX": "/opt/homebrew/opt/llvm@16/bin/clang++"
          }')
          || {}
          }}
```